app/javascript/components/workflows/workflow_payload.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Loading, Tabs, Tab,
+} from 'carbon-components-react';
+import AWSSfnGraph from '@tshepomgaga/aws-sfn-graph';
+import { Controlled as CodeMirror } from 'react-codemirror2';
+import '@tshepomgaga/aws-sfn-graph/index.css';
+
+const WorkflowPayload = ({ recordId }) => {
+  const tabLabels = [
+    { name: 'text', text: __('Text') },
+    { name: 'graph', text: __('Graph') },
+  ];
+  const [data, setData] = useState({
+    isLoading: true,
+    payload: '',
+    activeTab: tabLabels[0].name,
+  });
+
+  // TODO: Change the url when the GET automated worflow summary for recordId api is available.
+  useEffect(() => {
+    API.get(`/api/configuration_script_payloads/${recordId}?expand=resources`)
+      .then((response) => {
+        setData({
+          isLoading: false,
+          payload: response.payload,
+        });
+      });
+  }, [recordId]);
+
+  if (data.isLoading) {
+    return (
+      <div className="loadingSpinner">
+        <Loading active small withOverlay={false} className="loading" />
+      </div>
+    );
+  }
+
+  /** Function to set the active tab name */
+  const onTabSelect = (name) => {
+    setData({
+      ...data,
+      activeTab: name,
+    });
+  };
+
+  /** Function to render the code mirror component */
+  const error = false;
+  const renderCodeSnippet = () => (
+    <CodeMirror
+      className={`miq-codemirror workflow-payload ${error ? 'has-error' : ''}`}
+      options={{
+        mode: 'application/ld+json',
+        theme: 'eclipse',
+        lineNumbers: true,
+        lineWrapping: true,
+        foldGutter: true,
+        gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter', 'CodeMirror-lint-markers'],
+        autoCloseBrackets: true,
+        matchBrackets: true,
+        lint: true,
+      }}
+      style={{ height: '300px' }}
```

Comment:
Why is this set to 300px here but 800px in the css?

Reply:
had set the height back to `'auto'` since it will render a blank space for smaller payloads